src/request.ts

```diff
@@ -1235,6 +1236,73 @@ class DatastoreRequest {
     this.save(entities, callback!);
   }
 
+  merge(entities: Entities): Promise<CommitResponse>;
+  merge(entities: Entities, callback: SaveCallback): void;
+  /**
+   * Merge the specified object(s). If a key is incomplete, its associated object
+   * is inserted and the original Key object is updated to contain the generated ID.
+   * For example, if you provide an incomplete key (one without an ID),
+   * the request will create a new entity and have its ID automatically assigned.
+   * If you provide a complete key, the entity will be get the data from datastore
+   * and merge with the data specified.
+   * By default, all properties are indexed. To prevent a property from being
+   * included in *all* indexes, you must supply an `excludeFromIndexes` array.
+   *
+   * Maps to {@link Datastore#save}, forcing the method to be `merge`.
+   *
+   * @param {object|object[]} entities Datastore key object(s).
+   * @param {Key} entities.key Datastore key object.
+   * @param {string[]} [entities.excludeFromIndexes] Exclude properties from
+   *     indexing using a simple JSON path notation. See the examples in
+   *     {@link Datastore#save} to see how to target properties at different
+   *     levels of nesting within your entity.
+   * @param {object} entities.data Data to merge to the same for provided key.
+   * @param {function} callback The callback function.
+   * @param {?error} callback.err An error returned while making this request
+   * @param {object} callback.apiResponse The full API response.
+   */
+  merge(
+    entities: Entities,
+    callback?: SaveCallback
+  ): void | Promise<CommitResponse> {
+    const transaction = this.datastore.transaction();
+    transaction.run(err => {
+      if (err) {
+        transaction.rollback();
+        callback!(err);
+        return;
+      }
+      arrify(entities)
```

Comment:
Allowing multiple entities seems unnecessary. What do you think about just one per call to merge?

Reply:
@stephenplusplus Hi, I am following similar behavior from upsert and update function, it seems reasonable providing user to merge multiple entities.

Reply:
I see, thanks!